SEED: 21G5063a/MacBookPro11,4: Google Maps rendering (graphic distortion rendered against a black view where Maps should be)

grorg opened the pull request:
#### 8d2a69e734f034df7d5c79782eba867dcf63f348
<pre>
SEED: 21G5063a/MacBookPro11,4: Google Maps rendering (graphic distortion rendered against a black view where Maps should be)
<a href="https://bugs.webkit.org/show_bug.cgi?id=242764">https://bugs.webkit.org/show_bug.cgi?id=242764</a>
rdar://96780550

Reviewed by Kenneth Russell.

With the latest ANGLE Metal, older devices are seeing some issues. e.g. MacBook
Pros from 2015. They are no longer supported in Ventura, so this isn&apos;t an issue
there, but they are supported on Monterey. This fix forces them into using the
OpenGL backend for ANGLE.

* Source/WebCore/platform/graphics/cocoa/GraphicsContextGLCocoa.mm:
(WebCore::platformSupportsMetal): Return false if the device doesn&apos;t support MTLGPUFamilyMac2.
* Source/WebCore/testing/Internals.mm:
(WebCore::Internals::platformSupportsMetal):

Canonical link: <a href="https://commits.webkit.org/252480@main">https://commits.webkit.org/252480@main</a>
</pre>

kenrussell commented on Source/WebCore/testing/Internals.mm:
```diff
@@ -191,16 +191,21 @@ - (NSAttributedString *)_attributedStringForRange:(NSRange)range
 {
     auto device = adoptNS(MTLCreateSystemDefaultDevice());
 
+    UNUSED_PARAM(isWebGL2);
```
This is OK to use even if it's actually used later?

grorg replied:
Yeah, it is. That expands to something like `(void)isWebGL2;`

It's a bit ugly to do it this way, but the alternative was to have it in two of the three branches.

Thanks @kenrussell

kenrussell reviewed (approved):
Looks good to me. One question.

grorg committed:
SEED: 21G5063a/MacBookPro11,4: Google Maps rendering (graphic distortion rendered against a black view where Maps should be)
https://bugs.webkit.org/show_bug.cgi?id=242764
rdar://96780550

Reviewed by Kenneth Russell.

With the latest ANGLE Metal, older devices are seeing some issues. e.g. MacBook
Pros from 2015. They are no longer supported in Ventura, so this isn't an issue
there, but they are supported on Monterey. This fix forces them into using the
OpenGL backend for ANGLE.

* Source/WebCore/platform/graphics/cocoa/GraphicsContextGLCocoa.mm:
(WebCore::platformSupportsMetal): Return false if the device doesn't support MTLGPUFamilyMac2.
* Source/WebCore/testing/Internals.mm:
(WebCore::Internals::platformSupportsMetal):

Canonical link: https://commits.webkit.org/252480@main
